.github/workflows/smoke-test-release.yml

```diff
@@ -23,31 +23,31 @@ jobs:
             contents: read
         runs-on: ubuntu-20.04
         outputs:
-            tag: ${{ steps.tag.outputs.result }}
+            tag: ${{ steps.get-tag.outputs.tag }}
             created: ${{ steps.created-at.outputs.created }}
         steps:
             - name: Validate tag
               if: ${{ github.event_name == 'workflow_dispatch' }}
               env:
-                  GH_TOKEN: ${{ github.token }}
-              run: gh release view "${{ github.event.inputs.tag }}" --repo=woocommerce/woocommerce
+                  GH_TOKEN: ${{ secrets.E2E_GH_TOKEN }}
+              run: gh release view "${{ inputs.tag }}" --repo=woocommerce/woocommerce
 
-            - name: Get tag
-              uses: actions/github-script@v6
-              id: tag
-              with:
-                  result-encoding: string
-                  script: |
-                      console.log( "${{ github.event_name }}" );
-                      return "${{ github.event.release.tag_name }}" || "${{ github.event.inputs.tag }}"
+            - name: Get tag from triggered event
+              id: get-tag
+              env:
+                  RELEASE_TAG: ${{ github.event.release.tag_name || inputs.tag }}
+              run: |
+                  echo "Triggered event: ${{ github.event_name }}"
+                  echo "Tag from event: $RELEASE_TAG"
+                  echo "tag=$RELEASE_TAG" >> $GITHUB_OUTPUT
```

Comment:
[Fix]

The `return` line was erroneous because the `||` should be inside `${{ }}`. Fixed it here by doing just that.